### PR TITLE
Create an Enumerable module

### PR DESCRIPTION
**Pull request by mbabaliiryn**
**Add** new methods onto the existing Enumerable module

### CODE REVIEW

File: enumerable.rb [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+
+  def my_any?(_regx = nil)
+    my_each do |x|
+      return true if block_given? && yield(x)
+    end
+    false
+  end
+
+  def my_none?(_regx = nil)
+    my_each do |x|
```
Comment (Forison) [inline enumerable.rb]: * [ ] There should be working my_none? identical and returns the same thing as ruby's [`none?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-none-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_none.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
compare results in each example
```# p [1].none?(String) != [1].my_none?(String)```
```# p [1].none?(/\w/) != [1].my_none?(/\w/)```
```# p [1].none?(2) != [1].my_none?(2)```

File: enumerable.rb [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+  def my_all?(_regx = nil)
+    my_each do |x|
+      return false if block_given? && yield(x)
+    end
+    true
+  end
+
+  def my_any?(_regx = nil)
+    my_each do |x|
+      return true if block_given? && yield(x)
```
Comment (Forison) [inline enumerable.rb]: * [ ]  There should be a working `my_any?`  identical and returns the same thing as ruby's [`any?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-any-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_any.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
compare results returned in each case
``` p [1].any? != [1].my_any?```
``` p [1].any?(Integer) != [1].my_any?(Integer)```
``` p %w[a].any?(/\w/)  != %w[a].my_any?(/\w/) ```
``` p [1].any?(1)  != [1].my_any?(1) ```

File: enumerable.rb [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+    elsif block_given?
+      my_each { |x| counter += 1 if yield(x) }
+      counter
+    else
+      length
+    end
+  end
+
+  def my_map(&proc)
+    return to_enum(:my_map) unless block_given?
```
Comment (Forison) [inline enumerable.rb]: * [ ] check the previous issue raised with respect to ```my_map``` in the previous review

File: enumerable.rb [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+    end
+    num
+  end
+
+  def my_all?(_regx = nil)
+    my_each do |x|
+      return false if block_given? && yield(x)
+    end
+    true
+  end
```
Comment (suretrust) [inline enumerable.rb]: * [ ] should have working `my_all?` identical and returns the same thing as ruby's [`all?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-all-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_all.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
```ruby
[1,2,3,4,5,6].my_all?{ |num| num <= 6 } != [1,2,3,4,5,6].all?{ |num| num <= 6 }

[1,2,3,4,5,6].my_all?{ |num| num > 6 } != [1,2,3,4,5,6].all?{ |num| num > 6 }

[1,2,3,4,5,6].my_all?(Integer) != [1,2,3,4,5,6].all?(Integer)

%w[dog door rod blade].my_all?(/o/) != %w[dog door rod blade].all?(/o/)

[1,2,3,4,5,6].my_all?(3) != [1,2,3,4,5,6].all?(3)
```

File: enumerable.rb [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+
+  def my_any?(_regx = nil)
+    my_each do |x|
+      return true if block_given? && yield(x)
+    end
+    false
+  end
```
Comment (suretrust) [inline enumerable.rb]: * [ ] should have working `my_any?` identical and returns the same thing as ruby's [`any?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-any-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_any.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
```ruby
[nil, false, true, []].my_any? != [nil, false, true, []].any?

[1,2,3,4,5,6].my_any?(Integer) != [1,2,3,4,5,6].any?(Integer)

%w[dog door rod blade].my_any?(/d/) != %w[dog door rod blade].any?(/d/)

%w[dog door rod blade].my_any?('cat') != %w[dog door rod blade].any?('cat')
```

File: enumerable.rb [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+  def my_none?(_regx = nil)
+    my_each do |x|
+      if block_given?
+        return false if yield(x)
+      elsif x
+        return false
+      end
+    end
+    true
+  end
```
Comment (suretrust) [inline enumerable.rb]: * [ ] should have working `my_none?` identical and returns the same thing as ruby's [`none?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-none-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_none.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
```ruby
[1,2,3,4,5].my_none?(String) != [1,2,3,4,5].none?(String)

%w[dog door rod blade].my_none?(/z/) != %w[dog door rod blade].none?(/z/)

%w[dog door rod blade].my_none?(5) != %w[dog door rod blade].none?(5)
```

**Pull request merged**